### PR TITLE
{tools}[foss/2021b] snakemake v7.18.2 w/ Python 3.9.6

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3151.skitty.os - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/6faeea9301cfd8136227334bdc7962c0 for a full test report.

File: easybuild/easyconfigs/s/snakemake/snakemake-7.18.2-foss-2021b.eb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        # workaround for https://github.com/uburuntu/throttler/issues/3
+        'preinstallopts': "touch readme.md requirements-dev.txt && ",
+    }),
+    ('dpath', '2.0.6', {
+        'checksums': ['5a1ddae52233fbc8ef81b15fb85073a81126bb43698d3f3a1b6aaf561a46cdc0'],
+    }),
+    ('plac', '1.3.5', {
+        'checksums': ['38bdd864d0450fb748193aa817b9c458a8f5319fbf97b2261151cfc0a5812090'],
+    }),
+    # yte requires PyYAML >= 6.0
```
Comment (lexming) [inline easybuild/easyconfigs/s/snakemake/snakemake-7.18.2-foss-2021b.eb:66]: This is risky, loading another module combined with snakemake that depends on PyYAML v5.4.1 from this toolchain is not unlikely. In that case one might break the other depending on the load order.

Looking at the [changelog](https://github.com/yaml/pyyaml/blob/master/CHANGES) of PyYAML, one point of failure is that v6.0 made the `Loader` argument in `yaml.load()` mandatory.

I suggest to move this version of snakemake to `foss/2022a`.

**Review by lexming — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/s/snakemake/snakemake-7.18.2-foss-2021b.eb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+]
+
+use_pip = True
+sanity_pip_check = True
+
+exts_list = [
+    ('wrapt', '1.14.1', {
+        'checksums': ['380a85cf89e0e69b7cfbe2ea9f765f004ff419f34194018a6827ac0e3edfed4d'],
+    }),
+    ('ratelimiter', '1.2.0.post0', {
```
Comment (smoors) [inline easybuild/easyconfigs/s/snakemake/snakemake-7.18.2-foss-2021b.eb:25]: ratelimiter has been replaced with throttler, so can be removed?

File: easybuild/easyconfigs/s/snakemake/snakemake-7.18.2-foss-2021b.eb (hunk 1 above)
Comment (lexming) [reply easybuild/easyconfigs/s/snakemake/snakemake-7.18.2-foss-2021b.eb:66]: > but we take the same approach (installing a newer Python package as extension if needed) for various other tools as well

Indeed, and that's unreliable as a general approach, this is a good example for #15004

I won't block this PR though, as we don't have any clear guidelines in this regard.

File: easybuild/easyconfigs/s/snakemake/snakemake-7.18.2-foss-2021b.eb (hunk 2 above)
Comment (boegel) [reply easybuild/easyconfigs/s/snakemake/snakemake-7.18.2-foss-2021b.eb]: @smoors fixed in 164eb8f226

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @boegel!

**Pull request merged**